ta/remoteproc/remoteproc_core.c

```diff
@@ -0,0 +1,1212 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (C) 2023, STMicroelectronics - All Rights Reserved
+ */
+
+#include <elf_parser.h>
+#include <remoteproc_pta.h>
+#include <string.h>
+#include <sys/queue.h>
+#include <ta_remoteproc.h>
+#include <tee_internal_api.h>
+#include <tee_internal_api_extensions.h>
+#include <types_ext.h>
+#include <utee_defines.h>
+
+/*
+ *  signed file structure:
+ *
+ *                    ----+-------------+
+ *                   /    |    Magic    |
+ *                  /     +-------------+
+ *                 /      +-------------+
+ *                /       |   version   |
+ *               /        +-------------+
+ *              /         +-------------+
+ * +-----------+          | sign size   |   size of the signature
+ * |   Header  |          +-------------+   (in bytes, 64-bit aligned)
+ * +-----------+          +-------------+
+ *              \         | images size |   total size of the images to load
+ *               \        +-------------+   (in bytes, 64-bit aligned)
+ *                \       +-------------+
+ *                 \      |  TLV size   |   Generic TLV size
+ *                  \     +-------------+   (in bytes, 64-bit aligned)
+ *                   \    +-------------+
+ *                    \   | PLAT TLV sz |   Platform TLV size
+ *                     \--+-------------+   (in bytes, 64-bit aligned)
+ *
+ *                        +-------------+   Signature of the header, the trailer
+ *                        | Signature   |   and optionally the firmware image if
+ *                        +-------------+   a hash table is not stored.
+ *
+ *                        +-------------+
+ *                        |   Firmware  |
+ *                        |    image    |
+ *                        +-------------+
+ *                  ------+-------------+
+ *                 /      |+-----------+|
+ *                /       ||   TLV     ||   Information used to authenticate the
+ *  +-----------+/        ||           ||   firmware, stored in
+ *  |  Trailer  |         |+-----------+|   Type-Length-Value format.
+ *  +-----------+\        |+-----------+|
+ *                \       || Platform  ||   Specific platform information,
+ *                 \      ||   TLV     ||   stored in Type-Length-Value format.
+ *                  \     |+-----------+|
+ *                   -----+-------------+
+ *
+ *  TLV and platform TLV chunks:
+ *
+ *                        +-------------+
+ *                        |    Magic    |
+ *                        +-------------+
+ *                        +-------------+
+ *                        |    size     |  size of the TLV payload
+ *                        +-------------+
+ *                        +-------------+
+ *                        |     TLV     |  TLV structures
+ *                        |   payload   |
+ *                        +-------------+
+ */
+
+/* Firmware state */
+enum remoteproc_state {
+	REMOTEPROC_OFF,
```

Comment:
Prefer not having 2 enum values with the same name (`REMOTEPROC_OFF` from _core/pta/stm32mp/remoteproc_pta.c_) but that not a strong opinion as each is in the scope of its respective source file.

Reply:
I did not use the same name as the state enum could be different in the TA and the PTA, means we could have specific states in PTA depending on the platform implementation.